src/FindEmailAddress.java

```diff
@@ -0,0 +1,362 @@
+    {
+        if(inputString == null)
+            return null;
+
+        int point = 0;
+        int length = inputString.length();
+        while(point < length && inputString.charAt(point) == '/' )
+        {
+            point ++;
+        }
+        if(point<length)
+        {
+            return inputString.substring(point);
+        }else{
+            return "";
+        }
+
+    }
+
+
+    /*
+    *  check if this is a valid url we need to scrawl
+    */
+    private boolean checkIfSameDomainHtmlLink(String inputString)
+    {
+        boolean isSameDomainUrl = false;
+        if(inputString == null || inputString.trim().length()<=0)
+            return isSameDomainUrl;
+
+        String testString = inputString.trim();
+
+        if(!testString.toLowerCase().startsWith("http"))
+        {
+            if(!testString.startsWith("/"))
+                return isSameDomainUrl;
+
+            testString = removeExtraSlash(testString);
+            testString = "http://" + testString;
+        }else{
+            int parthStartIndex = testString.indexOf("//");
+            if(parthStartIndex <=0)
+            {
+                return isSameDomainUrl;
+            }
+
+            String protocolHeader = testString.substring(0,parthStartIndex+2);
+            testString = testString.substring(parthStartIndex+2);
+            if(testString.startsWith("/")) {
+                testString = removeExtraSlash(testString);
+            }
+            testString = protocolHeader + testString;
+        }
+
+        try {
+
+            URL inputURL = new URL(testString);
+            String host = inputURL.getHost();
+            String path = inputURL.getPath();
+
+
+            if(host == null)
+            {
+                host = this.requestDomain;
+            }else{
+                if(!host.equalsIgnoreCase(this.requestDomain))
+                {
+                    return isSameDomainUrl;
+                }
+
+            }
+
+            int dotIndex = path.lastIndexOf('.');
+            if (dotIndex >= 0) {
+                String fileExt = path.substring(dotIndex + 1);
+                if (fileExt.equalsIgnoreCase("html") || fileExt.equalsIgnoreCase("htm")) {
+                        isSameDomainUrl = true;
+                }
+
+            } else { // this mean this is a folder using default html page
+                   isSameDomainUrl = true;
+            }
+
+
+            if(testString.endsWith("/"))
+            {
+                testString = testString.substring(0,testString.length()-1);
+            }
+
+            //if this is a sameDomainURL, first check if it is already visited then add a new task to the thread pool
+            if(isSameDomainUrl)
+            {
+                //synchronized (urlHistory) {
+                    if (this.urlHistory.get(testString) == null && urlHistory.size() < maxUrlCheck) {
+
+                        this.urlHistory.put(testString, "");
```

Comment:
seems like you are only ever putting an empty string into this hash table; it should be a set instead.

Reply:
Agree, Set/HashSet should be used